### ISSUE TITLE
IOperation: Doesn't handle conversions of constant expressions

### ISSUE THREAD

**Issue by lgolding** [dotnet/roslyn#7299]
In this line:

```
private float f = 0.0;
```

... the `ConstantValue` property of the `IExpression` represented by the literal `0.0` is `null`. But in this line:

```
private double d = 0.0;
```

... the `ConstantValue` is `ConstantValueDoubleZero`.

The reason is that the compiler considered the first example as a "conversion expression" rather than a constant expression.

Similarly, the `ConstantValue` of this expression is `null` because the compiler considers it a conversion expression:

```
private char c = 'a' - 'a';
```

... but the `ConstantValue` of this expression is correctly `0`:

```
private char c = (char)('a' - 'a');
```

**Comment by JohnHamby** [comment]
This is a compiler issue that won't get fixed for 1.2.

**Comment by jinujoseph** [comment]
@333fred is this done ?
